nflow-explorer/src/app/workflow/workflow.js

```diff
@@ -4,15 +4,29 @@
   var m = angular.module('nflowExplorer.workflow', [
     'nflowExplorer.workflow.graph',
     'nflowExplorer.workflow.info',
-    'nflowExplorer.workflow.tabs'
+    'nflowExplorer.workflow.tabs',
+    'nflowExplorer.services.WorkflowService'
   ]);
 
   m.controller('WorkflowCtrl', function (workflow, definition,
-                                          parentWorkflow, childWorkflows) {
+      parentWorkflow, childWorkflows, $scope, config, $interval, WorkflowService) {
     var self = this;
     self.workflow = workflow;
     self.parentWorkflow = parentWorkflow;
     self.definition = definition;
     self.childWorkflows = childWorkflows;
+
+    function reloadWorkflow(workflowId) {
+      console.log('Fetching workflow id ' + workflowId);
+      WorkflowService.get(workflowId).then(function(workflow) {
+        self.workflow = workflow;
+      });
+    }
+    self.poller = $interval(function() { reloadWorkflow(self.workflow.id); },
+      config.radiator.pollPeriod * 1000);
+    $scope.$on('$destroy', function() {
```

Comment:
should we stop the poller if the workflow is in final state?

Reply:
No, because someone can move it manually to some non-final state? If someone wants to monitor finished workflow, I think we can grant him 1 rest api call every 15 seconds without overloading the servers :)